src/sentry/static/sentry/app/utils/tokenizeSearch.tsx

```diff
@@ -1,93 +1,176 @@
-import flatMap from 'lodash/flatMap';
-
 import {escapeDoubleQuotes} from 'app/utils';
 
-export type QueryResults = {
-  /**
-   * The text portion of the string query
-   */
-  query: string[];
-  [k: string]: string[];
+export enum TokenType {
+  OP,
+  TAG,
+  QUERY,
+}
+
+export type Token = {
+  type: TokenType;
+  key?: string;
```

Comment:
we usually add optional props to the end of the object 😉

Reply:
Putting the key after the value seems weird, but if you say so.